.infrastructure/helm-chart/templates/deployment.yml

```diff
@@ -0,0 +1,81 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Chart.Name }}-deployment
+  namespace: {{ .Values.namespace.todoapp.name }}
+spec:
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: {{ .Values.todoapp.rollingUpdate.maxSurge }}
+      maxUnavailable: {{ .Values.todoapp.rollingUpdate.maxUnavailable }}
+  selector:
+    matchLabels:
+      app: {{ .Chart.Name }}
+  template:
+    metadata:
+      labels:
+        app: {{ .Chart.Name }}
+    spec:
+      serviceAccountName: {{ .Chart.Name }}-secrets-reader
+      containers:
+      - name: todoapp
+        image: {{ .Values.image.todoapp.repository}}
```

Comment:
The same here, separately parametrize repository and tag

For example: {{ .Values.image.todoapp.repository}}:{{ .Values.image.todoapp.tag}}